src/beta_inc.jl

```diff
@@ -0,0 +1,886 @@
+            return a0*(case ? esum(mu,z) : exp(z))*(1.0 + SpecialFunctions.rgamma1pm1(b0))/t
+        end
+    else
+        z -= logbeta(a,b)
+        ans = case ? esum(mu,z) : exp(z)
+        return ans
+    end
+
+   
+    
+    if ans == 0.0
+        return 0.0
+    end
+    apb = a + b
+    if apb > 1.0
+        u = a + b - 1.0
+        z = (1.0 + SpecialFunctions.rgamma1pm1(u))/apb
+    else
+        z = 1.0 + SpecialFunctions.rgamma1pm1(apb)
+    end
+    c = (1.0 + SpecialFunctions.rgamma1pm1(a))*(1.0 + SpecialFunctions.rgamma1pm1(b))/z
+    return ans*(a0*c)/(1.0 + a0/b0)    
+end  
+
+"""
+    beta_inc_cont_fraction(a,b,x,y,lambda,epps)
+
+Compute ``I_{x}(a,b)`` using continued fraction expansion when a,b > 1.
+It is assumed that ``\\lambda = (a+b)*y - b``
+DLMF : https://dlmf.nist.gov/8.17#E22
+BFRAC(A,B,X,Y,LAMBDA,EPS)
+"""
+function beta_inc_cont_fraction(a::Float64, b::Float64, x::Float64, y::Float64, lambda::Float64, epps::Float64)
+    ans = brcmp1(0.0,a,b,x,y,false)
+    if ans == 0.0
+        return 0.0
+    end
+    c = 1.0 + lambda
+    c0 = b/a
+    c1 = 1.0 + 1.0/a
+    yp1 = y + 1.0
+
+    n = 0.0
+    p = 1.0
+    s = a + 1.0
+    an = 0.0
+    bn = 1.0
+    anp1 = 1.0
+    bnp1 = c/c1
+    r = c1/c
+    #CONT FRACTION
+
+    while true
+     n += 1.0
+     t = n/a
+     w = n*(b - n)*x
+     e = a/s
+     alpha = (p*(p+c0)*e*e)*(w*x)
+     e = (1.0 + t)/(c1 + 2*t)
+     beta = n + w/s +e*(c + n*yp1)
+     p = 1.0 + t
+     s += 2.0
+
+     #update an, bn, anp1, bnp1
+     t = alpha*an  + beta*anp1
+     an = anp1
+     anp1 = t
+     t = alpha*bn + beta*bnp1
+     bn = bnp1
+     bnp1 = t
+
+     r0 = r
+     r = anp1/bnp1
+     if abs(r - r0) <= epps*r
+        break
+     end
+     #rescale
+     an /= bnp1
+     bn /= bnp1
+     anp1 = r
+     bnp1 = 1.0
+    end
+    return ans*r
+end
+
+"""
+    beta_inc_asymp_exp(a,b,lambda,epps)
+
+Compute ``I_{x}(a,b)`` using asymptotic expansion for a,b >= 15.
+It is assumed that ``\\lambda = (a+b)*y - b``
+BASYM(A,B,LAMBDA,EPS)
+"""
+function beta_inc_asymp_exp(a::Float64, b::Float64, lambda::Float64, epps::Float64)
+    a0 = b0 = c = d = zeros(22)
+    e0 = 2/sqrt(pi)
+    e1 = 2^(-1.5)
+    sm = 0.0
+    ans = 0.0
+    if a > b
+        h = b/a
+        r0 = 1.0/(1.0 + h)
+        r1 = (b-a)/a
+        w0 = 1.0/sqrt(b*(1.0+h))
+    else
+        h = a/b
+        r0 = 1.0/(1.0 + h)
+        r1 = (b-a)/b
+        w0 = 1.0/sqrt(a*(1.0+h))
+    end
+    f = -a*SpecialFunctions.log1pmx(-(lambda/a)) - b*SpecialFunctions.log1pmx((lambda/b))
+    t = exp(-f)
+    if t == 0.0
+        return ans
+    end
+    z0 = sqrt(f)
+    z = 0.5*(z0/e1)
+    z² = 2.0*f
+
+    a0[1] = (2.0/3.0)*r1
+    c[1] = -0.5*a0[1]
+    d[1] = - c[1]
+    j0 = (0.5/e0)*erfcx(z0)
+    j1 = e1
+    sm = j0 + d[1]*w0*j1
+
+    s = 1.0
+    h² = h*h
+    hn = 1.0
+    w = w0
+    znm1 = z
+    zn = z²
+
+    for n = 2: 2: 20
+        hn *= h²
+        a0[n] = 2.0*r0*(1.0 + h*hn)/(n + 2.0)
+        np1 = n + 1
```

Comment:
I think it is easier to get rid of these variables: just use `n+1`. Compilers are pretty good at optimising this sort of thing